metricflow-semantics/metricflow_semantics/model/semantics/linkable_spec_resolver.py

```diff
@@ -711,12 +731,21 @@ def create_linkable_element_set_from_join_path(
         join_path: SemanticModelJoinPath,
     ) -> LinkableElementSet:
         """Given the current path, generate the respective linkable elements from the last semantic model in the path."""
+        semantic_model = self._semantic_model_lookup.get_by_reference(join_path.last_semantic_model_reference)
+        assert semantic_model
```

Comment:
Can you add an error message for this assertion? Something that tells us that something is internally misconfigured, not user error. We tend to hit these errors more than we expect, so trying to get in the habit of always including a message.